pkg/oauthflow/flow.go

```diff
@@ -88,3 +83,30 @@ func emailFromIDToken(tok *oidc.IDToken) (string, error) {
 	}
 	return claims.Email, nil
 }
+
+type StaticTokenGetter struct {
+	RawToken string
+}
+
+func (stg *StaticTokenGetter) GetIDToken(_ *oidc.Provider, _ oauth2.Config) (*OIDCIDToken, error) {
+	unsafeTok, err := jose.ParseSigned(stg.RawToken)
+	if err != nil {
+		return nil, err
+	}
+	// THIS LOGIC IS GENERALLY UNSAFE BUT OK HERE
+	// We are only parsing the id-token passed directly to a command line tool by a user, so it is trusted locally.
+	// We need to extract the email address to attach an additional signed proof to the server.
+	// THE SERVER WILL DO REAL VERIFICATION HERE
+	unsafePayload := unsafeTok.UnsafePayloadWithoutVerification()
```

Comment:
We just need to get out the email address since we have to sign it for the challenge. The alternative would just be to accept both on the command line.

Reply:
To expand: to actually verify we'd need to construct a full verifier, which needs the oauth config to fetch keys, etc. We don't have all that info necessarily in the client.